### PR TITLE
add article fetchers to content app

### PR DESCRIPTION
**Pull request by jamesgorrie**
* Moves fetching into content app
* Stops using prismic client to fetch on the client side, rather uses Next [api routes](https://nextjs.org/docs/api-routes/introduction)
* Still uses `common/prismic/parsers` <= because of the multiple shapes of our components and their props, this might be a bigger fish...

There's some patterns forming in the fetch functions, but I'd rather keep it copy pastey until we have more use cases.

### CODE REVIEW

File: common/utils/array.ts [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+export function isUndefined<T>(val: T | undefined): val is T {
+  return typeof val === 'undefined';
+}
+
+export function isString(v: any): v is string {
+  if (typeof v === 'string') {
+    return true;
+  }
+  return false;
+}
```
Comment (jamesgorrie) [inline common/utils/array.ts:14]: This file is oddly named, I'll rename it as another PR.

File: content/webapp/services/prismic/fetch/articles.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { Query } from '@prismicio/types';
+import { GetServerSidePropsPrismicClient } from '.';
+import { ArticlePrismicDocument } from '../articles';
+
+const fetchLinks = [
```
Comment (jamesgorrie) [inline content/webapp/services/prismic/fetch/articles.ts:5]: https://prismic.io/docs/technologies/fetch-linked-document-fields-javascript

File: common/utils/array.ts [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+
+export function isUndefined<T>(val: T | undefined): val is T {
+  return typeof val === 'undefined';
+}
+
+export function isString(v: any): v is string {
+  if (typeof v === 'string') {
+    return true;
+  }
+  return false;
```
Comment (alexwlchan) [inline common/utils/array.ts:13]:
```suggestion
  return typeof v === 'string';
```

?

**Review by alexwlchan — APPROVED** [review]
🍝

**Pull request merged**